File: Assets/Bundles/Server/Resources/TS/Services/Global/Chat/Commands/DamageCommand.lua
-- Compiled with unity-ts v2.1.0-75
local Flamework = require("Shared/rbxts_include/node_modules/@easy-games/flamework-core/out/init").Flamework
local ChatCommand = require("Server/TS/Services/Global/Chat/Commands/ChatCommand").ChatCommand
local DamageCommand
do
	local super = ChatCommand
	DamageCommand = setmetatable({}, {
		__tostring = function()
			return "DamageCommand"
		end,
		__index = super,
	})
	DamageCommand.__index = DamageCommand
	function DamageCommand.new(...)
		local self = setmetatable({}, DamageCommand)
		return self:constructor(...) or self
	end
	function DamageCommand:constructor()
		super.constructor(self, "damage")
	end
	function DamageCommand:Execute(player, args)
		local amount
		local target
		if #args == 1 then
			amount = tonumber(args[1])
			target = (Flamework.resolveDependency("Bundles/Server/Services/Global/Entity/EntityService@EntityService")):GetEntityByClientId(player.clientId)
		end
		if amount == nil then
			player:SendMessage("invalid amount: " .. tostring(amount))
			return nil
		end
		if target == nil then
			player:SendMessage("invalid target")
			return nil
		end
		(Flamework.resolveDependency("Bundles/Server/Services/Global/Damage/DamageService@DamageService")):InflictDamage(target, amount, {
			ignoreCancelled = true,
		})
		player:SendMessage("Inflicted " .. (tostring(amount) .. (" dmg to " .. tostring(target.id))))
	end
end
return {
	DamageCommand = DamageCommand,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Server/Services/Global/Chat/Commands/DamageCommand.ts
import { Dependency } from "@easy-games/flamework-core";
import { DamageService } from "Server/Services/Global/Damage/DamageService";
import { EntityService } from "Server/Services/Global/Entity/EntityService";
import { Entity } from "Shared/Entity/Entity";
import { Player } from "Shared/Player/Player";
import { ChatCommand } from "../../../../Commands/ChatCommand";

export class DamageCommand extends ChatCommand {
	constructor() {
		super("damage");
	}

	public Execute(player: Player, args: string[]): void {
		let amount: number | undefined;
		let target: Entity | undefined;

		if (args.size() === 1) {
			amount = tonumber(args[0]);
			target = Dependency<EntityService>().GetEntityByClientId(player.clientId);
		}

		if (amount === undefined) {
			player.SendMessage("invalid amount: " + amount);
			return;
		}

		if (target === undefined) {
			player.SendMessage("invalid target");
			return;
		}

		Dependency<DamageService>().InflictDamage(target, amount, {
			ignoreCancelled: true,
		});
		player.SendMessage(`Inflicted ${amount} dmg to ${target.id}`);
	}
}
